erdpy/simulation.py

```diff
@@ -17,7 +25,7 @@ def to_dictionary(self) -> Dict[str, Any]:
         return dictionary
 
 class Simulator():
-    def __init__(self, proxy: IElrondProxy) -> None:
+    def __init__(self, proxy: INetworkProvider) -> None:
```

Comment:
Maybe drop everything related to `simulate_transaction_cost`?

If I remember well, this function is not implemented in eggs-network-providers.